### PR TITLE
Add desktop panning with mouse events

### PR DESCRIPTION
**Pull request by cathyxz**
Closes https://github.com/ampproject/amphtml/issues/17845

TODO: 
- [x] mobile regression testing on actual devices
- [x] add automated unit tests

### CODE REVIEW

File: extensions/amp-pan-zoom/0.1/amp-pan-zoom.js [hunk 1]
```diff
@@ -395,25 +407,105 @@ export class AmpPanZoom extends AMP.BaseElement {
...
+    this.unlistenMouseUp_ =
+        listen(this.win, 'mouseup', this.onMouseUp_.bind(this));
+  }
+
+  /**
+   * @param {Event} e
+   * @private
+   */
+  onMouseMove_(e) {
+    e.preventDefault();
```
Comment (sparhami) [inline extensions/amp-pan-zoom/0.1/amp-pan-zoom.js]: Might want to consider adding a comment what the preventDefault does, not immediately obvious to me.

File: extensions/amp-pan-zoom/0.1/amp-pan-zoom.js [hunk 2]
```diff
@@ -751,9 +848,11 @@ export class AmpPanZoom extends AMP.BaseElement {
       if (this.scale_ <= 1) {
```
Comment (sparhami) [inline extensions/amp-pan-zoom/0.1/amp-pan-zoom.js]: shouldn't this be `if (this.scale_ <= this.minScale_)`?

Comment (cathyxz) [reply extensions/amp-pan-zoom/0.1/amp-pan-zoom.js]: Yeah you're right. I was under the impression that we fixed this in a previous PR but maybe somewhere a bad merge happened. 0.0

File: extensions/amp-pan-zoom/0.1/amp-pan-zoom.js [hunk 3]
```diff
@@ -395,25 +407,105 @@ export class AmpPanZoom extends AMP.BaseElement {
...
+   */
+  cleanupEvents_() {
+    this.cleanupGestures_();
+    this.unlisten_(this.unlistenMouseDown_);
+    this.unlisten_(this.unlistenMouseMove_);
+    this.unlisten_(this.unlistenMouseUp_);
+  }
+
+  /**
+   * Mouse down handler for panning in desktop mode
```
[the remark this answers is not in the document]
Comment (cathyxz) [reply extensions/amp-pan-zoom/0.1/amp-pan-zoom.js]: Done below in `onZoomRelease`.

**Comment by cathyxz** [comment]
@sparhami thanks a lot for the thorough review. PTAL?

**Comment by cathyxz** [comment]
Oh github race condition. Thanks!

**Pull request merged**